### ISSUE TITLE
66 unit tests failed in OC Catalyst project in the dev branch

### ISSUE THREAD

**Issue by smitbmx** [ordercloud-api/ordercloud-dotnet-catalyst#47]
1. Need to understand the reason for the recent change in the _GlobalExceptionHandler_ and which benefits it brings.
2. Fix tests or revert/reimplement the commit d466808536f48f1e81ca92c3f168d9b9c8a19fa0 [d466808].
![image](https://user-images.githubusercontent.com/1814356/147091287-aead89d5-112d-43d9-8bcc-8bd63965f2b2.png)

**Comment by smitbmx** [comment]
I believe the mentioned change should be reverted which will make the `dev` branch green aagain. We don't need this change until we got a request from the community why we need this (and I believe the implementation should be improved due to not expected exception type).
This change makes clients to implement more complicated solutions on their side and forces to add additional `try-catch` construction. The `throw` shows not the relevant exception in the first place, for example, if we have a request with a wrong address or any other we got the: `Error while copying content to a stream` exception. 
Clients can utilize the `GetBaseException` method on the mentioned exception variable, cast to the `CatalystBaseException` type, and only after that retrieve the relevant information. 
Prototype code of how to fix the unit-test or what should have clients on their side:
```
try
{
    await TestFramework.Client.Request("demo/notfound").GetAsync();
}
catch (Exception e)
{
    CatalystBaseException baseException = e.GetBaseException() as CatalystBaseException;
    baseException.ShouldHaveFirstApiError("NotFound", 404, "Not found.");
}
```
If we revert the `throw` from the handler, the code style will be as before, much simpler:
```
var result = await TestFramework.Client.Request("demo/notfound").GetAsync();
await result.ShouldHaveFirstApiError("NotFound", 404, "Not found.");
```
